Automattic/wp-calypso
# Jetpack Manage: 134 - add quick links nav to overview sidebar

Resolves Automattic/jetpack-manage#134 - add quick links nav to overview sidebar

## Proposed Changes
This implements a `FoldableNav` component, which extends `FoldableCard` and builds out a simple list of links. It also ensures the `expanded` setting is saved.

The `QuickLinksNav` uses this new component, and lists links as per the specs:

* Design: pf36In-fY-p2
* Technical breakdown: pf36In-mf-p2

![image](https://github.com/Automattic/wp-calypso/assets/32492176/20c2926c-b499-42c2-b366-68ca426a16b8)


## Testing Instructions
* Review the code.
* Verify the component matches the provided design.
* Verify the `expanded` state is preserved upon refresh.
* Verify tracks events fire when clicking a link.
* Verify tracks events fire when expanding/collapsing the component.

Note that there is quite a bit of mock content in the parent components; please ignore layout quirks outside of this component for now.

Once this is merged, I've also got a `GetHelpNav` component ready to go (pending any changes needed here).
## Pre-merge Checklist
